### ISSUE TITLE
[Bug] Does not work as expected - firefox

### ISSUE THREAD

**Issue by arimal199** [darkreader/darkreader#11264]
### Prerequisites

- [X] I [searched for any existing report](https://github.com/darkreader/darkreader/issues?q=is%3Aissue) about this bug to avoid opening a duplicate.
- [X] I can reproduce this bug in a new, unmodified web browser profile with Dark Reader installed as the only extension.
- [X] I understand I need to use the [Broken Website Report template](https://github.com/darkreader/darkreader/issues/new?assignees=&labels=Broken+Website&template=broken-website-report.yml&title=%5BBroken+Website%5D+Replace+with+title) if this bug I am reporting occurs on a single website.

### Bug Description

Firefox extension is just broken. no matter the os nor the configuration - when using automation, the dark mode just won't turn on.

### Website Address

all

### Steps To Reproduce

enable automation on firefox addon

### Expected Behavior

automation is enabled

### Actual Behavior

light mode persists

### Screenshots

<img width="524" alt="image" src="https://github.com/darkreader/darkreader/assets/16977090/edd470d9-6a48-48c0-bc5d-61550369d786">

### Operating System

mac, linux, windows

### Web Browser name and version

firefox 113

### Dark Reader version

4.9.63

**Comment by muradbu** [comment]
In my years of using Dark Reader with Firefox, Chrome and Safari across Windows, Linux and macOS the automation feature never worked sadly.

**Comment by antebudimir** [comment]
> In my years of using Dark Reader with Firefox, Chrome and Safari across Windows, Linux and macOS the automation feature never worked sadly.

Yep, been using it for years as well and the same thing for me. Always had to poke at it manually.

**Comment by allen-liaoo** [comment]
same, I'm on mac and it never worked for me, no matter on firefox or chrome

**Comment by alexanderby** [comment]
Maybe the checkbox is displaying a wrong state? Please try unchecking and checking it again.

**Comment by muradbu** [comment]
> Maybe the checkbox is displaying a wrong state? Please try unchecking and checking it again.

Checking/unchecking the checkbox toggles dark mode if it's supposed to. But the automation doesn't work at all, I have to do it manually every time.

**Comment by antebudimir** [comment]
On my phone (Firefox), automation, i.e. from 20h to 06h, works.
On my laptop (Firefox), it turns on automatically, but I have to manually turn it off in the morning.

**Comment by antebudimir** [comment]
Recently, it started working as expected! Like 2-3 weeks ago! Nice work!

**Comment by muradbu** [comment]
I found my own workaround to getting automation to work.

Instead of using Dark Reader's built-in timer I checked "use system color scheme" and I let macOS toggle light/dark mode depending on sunset and sunrise. This works quite well, and it's also more unified because it changes with the OS' dark/light theme, dynamic wallpaper and Night Shift. 

I'm sure you can achieve the same on other OS'.
